packages/yew/src/html/component/scope.rs

```diff
@@ -222,10 +227,26 @@ mod feat_ssr {
             );
             scheduler::start();
 
+            if hydratable {
+                #[cfg(debug_assertions)]
+                w.push_str(&format!("<!--<[{}]>-->", std::any::type_name::<COMP>()));
```

Comment:
I'm a bit allergic to unescaped printing of html, but I think in this case it's fine, per the docs of `type_name`.

I know hydration isn't part of *this* PR, but can you explain how the comments node help, technically? Is it checking that the typenames match and the `[]` is used to indicated an element, vs `?` indicating a suspension, etc? This could be a type that codifies this encoding/decoding along the lines of
```rust
enum HydrationKind {
    Suspension,
    Component(String),
}
impl HydrationKind {
    fn from_component<COMP: BaseComponent>() -> Self { .. }
    fn suspension() -> Self { ... }
    fn render_start(&self, w: &mut String) {}
    fn render_end(&self, w: &mut String) {}
}
```

Reply:
> I know hydration isn't part of this PR, but can you explain how the comments node help, technically?

When a component is hydrating, it needs to know how many nodes to take before the next component / element can hydrate. This requires a `VNode`, which is not available until the component is first rendered.
Similarly, this information is not available until the first time all component becomes unsuspended for a Suspense boundary.

Assistive nodes allow the content of a component / Suspense to be collected immediately without a `VNode`.

This also allows additional state information to be carried to the client-side at a per-component level (e.g.: this makes `getServerSideProps`-like usage possible without having to print it outside of the ssr artifact, not included in the initial hydration pull request, but should come before 0.20).

> Is it checking that the typenames match

No, this is only printed in debug mode and intended for developers to debugging hydration layout mismatch. which is also why it's not escaped as it makes the type name hard to read and defeats the purpose.

It only needs `<!--<[]>--><!--</[]>-->` (the release mode output) to collect a fragment.